lib/ohai/plugins/virtualbox.rb

```diff
@@ -0,0 +1,44 @@
+#
+# Author:: "Tim Smith" <tsmith@chef.io>
+# Copyright:: Copyright (c) 2015 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+Ohai.plugin(:Virtualbox) do
+  provides 'virtualbox'
+
+  collect_data do
+    so = shell_out('VBoxControl guestproperty enumerate')
+    if so.exitstatus == 0
```

Comment:
usually i'd do `return unless so.exitstatus == 0`.  avoids indenting all the code.  violates the principle that 'there should be only one return' but i always violate that for sanity checks at the top of a method.

i think you may also be able to just shell_out! and let ohai fail the plugin.